File: backend/src/modules/routing/domain/value-objects/routing-profile.ts
/** Transport profiles exposed by our API (independent from any routing engine). */
export const ROUTING_PROFILES = ['CAR', 'TRUCK', 'MOTORCYCLE', 'BICYCLE', 'PEDESTRIAN'] as const;

export type RoutingProfile = (typeof ROUTING_PROFILES)[number];

export const isRoutingProfile = (value: unknown): value is RoutingProfile =>
  typeof value === 'string' && (ROUTING_PROFILES as readonly string[]).includes(value);

File: mobile/lib/domain/entities/routing_profile.dart
/// Transport profiles supported by the routing API (`profile` field).
enum RoutingProfile {
  car('CAR', 'Auto'),
  truck('TRUCK', 'Camión'),
  motorcycle('MOTORCYCLE', 'Moto'),
  bicycle('BICYCLE', 'Bicicleta'),
  pedestrian('PEDESTRIAN', 'A pie');

  const RoutingProfile(this.apiValue, this.label);

  /// Value sent to and received from the API.
  final String apiValue;

  /// Spanish label for the UI.
  final String label;

  static RoutingProfile fromApi(String value) {
    for (final profile in values) {
      if (profile.apiValue == value) return profile;
    }
    throw FormatException('Unknown routing profile "$value"');
  }
}
